---
title: "ATAC_NFIAB_KO"
output: github_document
---

```{r setup, include=FALSE}

knitr::opts_chunk$set(echo = TRUE)

```

# ATAC analysis
Differential accessibility defects in NIFA/B KO. 

```{r message=FALSE}

rm(list=ls())

library(DESeq2)
library(RColorBrewer)
library(tidyverse)
library(ComplexHeatmap)

```

### Load settings

Colors, main directory 

```{r}
source('./r_inputs/TemporalSpatialNeuralTube_settings.R')
```


### Set dirs
```{r}
outdir="outputs_glialatac_6_NFIAB_KO/"
subworkinput="inputs_glialatac_1_eda_pca/"
ifelse(!dir.exists(file.path(workingdir,outdir)), dir.create(file.path(workingdir,outdir)), "Directory exists")


```



## Load data

```{r }

#counts table
count_table <- read.table(file=paste0(workingdir,subworkinput,"consensus_peaks.mLb.clN.featureCounts.txt"),header=TRUE, stringsAsFactors = FALSE)

# clean colnames
colnames(count_table) <- gsub(".mLb.clN.bam","",colnames(count_table))

# we do not need coordinates
count_table <- count_table %>%
  select("Geneid", starts_with(c("MUT","WT")))


## Annotation table
ann_table <- read.table(file=paste0(workingdir,subworkinput,"consensus_peaks.mLb.clN.annotatePeaks.txt"), header=TRUE, stringsAsFactors = FALSE, sep = "\t")
colnames(ann_table)[1] <- "Peakid"


ann_table_clean <- ann_table %>% 
  select(c("Peakid","Chr","Start","End","Strand","Annotation","Distance.to.TSS","Nearest.PromoterID")) %>%
  separate(Annotation, into = "Annotation_brief", sep = " ", remove = FALSE)


```




## Differential analysis between WT and KO for each domain at two timepoints

Targeted diff analysis in subsets of samples D11:

WT vs KO: pairwise for p1, p2, pMN



```{r diff-KO-WT}


#subset 
timepoint=c("_D7_","_D11_")

#subset 
allgates=c("_p2_","_pM_","_p1_")


count_matrix <- count_table %>%
  column_to_rownames("Geneid")


PairWiseDEseq <- lapply(c(1:length(timepoint)),function (x) {
  lapply(c(1:length(allgates)), function (y) {
      timepoints <- timepoint[x]
      celltypes <- allgates[y]
      sub_counts <- count_matrix %>%
        dplyr::select(contains(celltypes) & contains(timepoints))
      
      ## Make metadata file for DESeq
      genecolData_sub <- data.frame(Sample_ID = colnames(sub_counts))
      genecolData_sub <- genecolData_sub %>% 
        separate(Sample_ID,into=c("Genotype","Day","Gate","NFIAgate","Rep"), sep="_", remove=FALSE) %>%
        mutate(Condition=paste(Genotype,Day,Gate,NFIAgate, sep="_"),
         DayNFIA=paste(Day,NFIAgate,Genotype,sep = "_"),
         DayGate=paste(Day,Gate,sep="_"),
         Experiment=paste(Genotype,Rep,sep="_"),
         NFIAstatus=paste(NFIAgate,Genotype,sep="_"))
      genecolData_sub <- as.data.frame(unclass(genecolData_sub))
      
      dds_sub <- DESeqDataSetFromMatrix(countData = sub_counts,
                                    colData = genecolData_sub,
                                    design = ~ Genotype)
      
      dds_sub <- DESeq(dds_sub)
      
      vsd_sub <- varianceStabilizingTransformation(dds_sub,blind = FALSE)
      
      # Export normalized tables for plotting elsewhere
      dds_sub_counts <- counts(dds_sub, normalized = TRUE)
      vsd_sub_data <- assay(vsd_sub)
      
      results_sub <- results(dds_sub)

      results_return <- results_sub %>% as.data.frame() %>% rownames_to_column("Geneid")
      results_return$Comparison <- paste0("Comp_WT_KO",timepoints,celltypes)
      results_return

  })
}) 

```

### How many diff acc elements between WT and KO?

D7 WT vs KO does not have any elements fulfilling the filtering so they don't get plotted. TODO : change the bar graph to plot zeros by using something like count. 

```{r filter-KO}
list <- unlist(listoflists, recursive = FALSE)
results_deseq_nfia <- do.call(rbind,unlist(PairWiseDEseq))

top_KO_comparisons <- results_deseq_nfia %>%
  as.data.frame() %>%
  filter(padj < 0.01 & abs(log2FoldChange) > 2 & baseMean > 100) 

top_KO_comparisons_count <- results_deseq_nfia %>%
  as.data.frame() %>%
  filter(padj < 0.01 & abs(log2FoldChange) > 2 & baseMean > 100) %>%
  mutate(Comparison=factor(Comparison, levels=results_deseq_nfia$Comparison %>% unique())) %>%
  group_by(Comparison) %>% dplyr::count(Comparison,.drop=FALSE)


ggplot(top_KO_comparisons_count, aes(x=Comparison,y=n)) +
  geom_col() +
  geom_point(color="black", fill="grey",shape=21) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))


```

### Panel: upregulated with KO changes

```{r plot-vsd-uptime}

interval_subset_up <- top_days_comparisons %>% filter(log2FoldChange < 0) %>%
  select(Intervals) %>% unique()
#  top_days_comparisons$Intervals %>% unique()



# filter elements
vsd_hm <- count_vsd %>%
  filter(X %in% interval_subset_up$Intervals) %>%
  column_to_rownames("X")


dim(vsd_hm)

# z score
vsd_hm_z <- t(scale(t(vsd_hm))) 

# # order columns as previous heatmap - but in reverse
# vsd_hm_z <- vsd_hm_z[,rev(col_order)]

vsd_hm_z <- vsd_hm_z[,sorted.sample.wReps.all]



# annotate the rows for which elements are affected by KO : top_KO_comparisons
row_KO_annotation <- data.frame(intervalid=row.names(vsd_hm_z)) %>%
  mutate(KO_diffacc=case_when(intervalid %in% top_KO_comparisons$Intervals ~ "diffacc", TRUE ~ "unchanged")) %>%
  select("KO_diffacc")

ann_color_row_JD <- list(
  KO_diffacc = c(diffacc="#5a5a5a",unchanged="white"))



rowAnn <- HeatmapAnnotation(
    df = row_KO_annotation,
    which = 'row', # 'col' (samples) or 'row' (gene) annotation?
    na_col = 'white', # default colour for any NA values in the annotation data-frame, 'ann'
    col = ann_color_row_JD,
    annotation_height = 0.6,
    annotation_width = unit(1, 'cm'),
    gap = unit(1, 'mm'))


# metadata for the heatmap
genecolData_first <- data.frame(Sample_ID = colnames(vsd_hm_z))
genecolData_first <- genecolData_first %>% 
  separate(Sample_ID,into=c("Genotype","Day","Gate","NFIAgate","Rep"), sep="_", remove=FALSE) %>%
  mutate(Condition=paste(Genotype,Day,Gate,NFIAgate, sep="_"),
         DayNFIA=paste(Day,NFIAgate,Genotype,sep = "_"),
         DayGate=paste(Day,Gate,sep="_"),
         Experiment=paste(Genotype,Rep,sep="_"),
         NFIAstatus=paste(NFIAgate,Genotype,sep="_"))
genecolData_first <- as.data.frame(unclass(genecolData_first))

phen_data <- genecolData_first %>%
  select(c("Sample_ID","Day","DayGate","NFIAstatus")) %>%
  remove_rownames() %>%
  column_to_rownames("Sample_ID")

ann_color_JD <- list(
  DayGate = c(D5_p1="#abdff4",D5_p2="#f1df9a", D5_pM="#f19aac",
              D7_p1="#55bee8",D7_p2="#e6c444",D7_pM="#e64466",
              D9_p1="#1a91c1",D9_p2="#c19e1a",D9_pM="#c11a3d",
              D11_p1="#0e506b",D11_p2="#6b570e",D11_pM="#7c1127"),
  NFIAstatus = c(NFIAn_WT="#f6f6f6",NFIAp_WT="#e8e8e8",`100`="#808080",NFIAn_MUT="#595959"),
  Day = c(D5="#fadede",D7="#f3aaaa",D9="#e96666",D11="#cf1e1e"))



# Annotated heatmap with selected colors
hm_colors = colorRampPalette(rev(brewer.pal(n = 11, name = "RdBu")))(100)


# Build the annotation for the complex heatmap
colAnn <- HeatmapAnnotation(
    df = phen_data,
    which = 'col', # 'col' (samples) or 'row' (gene) annotation?
    na_col = 'white', # default colour for any NA values in the annotation data-frame, 'ann'
    col = ann_color_JD,
    annotation_height = 0.6,
    annotation_width = unit(1, 'cm'),
    gap = unit(1, 'mm'))



hmap <- Heatmap(vsd_hm_z,

    name = 'Z-score',

    col = hm_colors,
    # split by KO or not
      row_split = row_KO_annotation,

    # row (gene) parameters
      cluster_rows = TRUE,
      show_row_dend = TRUE,
      #row_title = 'Statistically significant genes',
      row_title_side = 'left',
      row_title_gp = gpar(fontsize = 12,  fontface = 'bold'),
      row_title_rot = 90,
      show_row_names = FALSE,
      row_names_gp = gpar(fontsize = 10, fontface = 'bold'),
      row_names_side = 'left',
      row_dend_width = unit(25,'mm'),

    # column (sample) parameters
      cluster_columns = FALSE,
      show_column_dend = TRUE,
      column_title = '',
      column_title_side = 'bottom',
      column_title_gp = gpar(fontsize = 12, fontface = 'bold'),
      column_title_rot = 0,
      show_column_names = FALSE,
      column_names_gp = gpar(fontsize = 8),
      column_names_max_height = unit(10, 'cm'),
      column_dend_height = unit(25,'mm'),

    # cluster methods for rows and columns
      clustering_distance_columns = function(x) as.dist(1 - cor(t(x))),
      clustering_method_columns = 'ward.D2',
      clustering_distance_rows = function(x) as.dist(1 - cor(t(x))),
      clustering_method_rows = 'ward.D2',

    # specify top and bottom annotations
    right_annotation = rowAnn,  
    top_annotation = colAnn)

```


```{r fig.width=6.75, fig.height=3.5}

draw(hmap,
    heatmap_legend_side = 'left',
    annotation_legend_side = 'left',
    row_sub_title_side = 'left')

# print heatmap
pdf(paste0(workingdir,subworkinput,"Heatmap_NFIAB_KOup_ATAC.pdf"), width = 6.75, height = 3.5) 

draw(hmap,
    heatmap_legend_side = 'left',
    annotation_legend_side = 'left',
    row_sub_title_side = 'left')

dev.off()

```



```{r}
sessionInfo()
```