---
title: "Nr6a1 overexpression day 9 ATAC"
output: github_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Plottinng Nr6a1 overexpression results

```{r message=FALSE}

rm(list=ls())
library(DESeq2)
library(RColorBrewer)
library(tidyverse)
library(ggrastr)

```

### Load settings

Colors, main directory 

```{r}

source('./r_inputs/TemporalSpatialNeuralTube_settings.R')

```

### Set dirs

```{r}

subworkinput="inputs_glialnr6a1_2/"
outdir="outputs_glialnr6a1_2/"
ifelse(!dir.exists(file.path(workingdir,outdir)), dir.create(file.path(workingdir,outdir)), "Directory exists")

```

## Load data

Counts table generated by nf-core/atacseq in: `results/bwa/mergedLibrary/macs/broadPeak/consensus/`

```{r }

#counts table
count_table_full <- read.table(file=paste0(workingdir,subworkinput,"consensus_peaks.mLb.clN.featureCounts.txt"),header=TRUE, stringsAsFactors = FALSE)


# clean colnames
colnames(count_table_full) <- gsub(".mLb.clN.bam","",colnames(count_table_full))


# we do not need coordinates
count_table <- count_table_full %>%
  select("Geneid", Nr6a1OE_R2:Nr6a1OE_R3)


```


## Differential analysis

```{r}

count_matrix <- count_table %>%
  column_to_rownames("Geneid")

## Make metadata file for DESeq

genecolData_first <- data.frame(Sample_ID = colnames(count_matrix))
genecolData_first <- genecolData_first %>% 
  separate(Sample_ID,into=c("Condition","Rep"), sep="_", remove=FALSE)
genecolData_first <- as.data.frame(unclass(genecolData_first))


dds <- DESeqDataSetFromMatrix(countData = count_matrix,
                              colData = genecolData_first,
                              design = ~ Condition)

dds <- DESeq(dds)

vsd <- varianceStabilizingTransformation(dds,blind = FALSE)

```

## Export files 

Export normalized tables for plotting elsewhere

```{r }

dds_counts <- counts(dds, normalized = TRUE)
vsd_data <- assay(vsd)
dds_results <- results(dds)

# write.table(dds_counts, file = "1_output_tables/consensus_peaks.mLb.clN.normCounts.txt", quote = FALSE, row.names = TRUE)
# write.csv(vsd_data, file = "1_output_tables/consensus_peaks.mLb.vsd.csv", quote = FALSE)

```



## Volcano plot


```{r}
# color significant
results_sub_plot1 <- dds_results %>%
  as.data.frame() %>%
  mutate(color_sig=case_when(padj < 0.1 & padj > 0.01 ~ "under01",
                      padj < 0.01 & padj > 0.001 ~ "under001",
                      padj < 0.001 & padj >0 ~ "under0001",
                      TRUE ~ "over01"))

```



```{r, fig.width=4, fig.height=3}

threshold <- 2

ggplot(results_sub_plot1, aes(x=log2FoldChange, y=-log10(padj), color=color_sig)) +
          geom_point(size=1) +
          geom_point(data=results_sub_plot1[results_sub_plot1$log2FoldChange > threshold,],
                   aes(y=-log10(padj), x=threshold), shape = 2, colour="gray30") +
          geom_point(data=results_sub_plot1[results_sub_plot1$log2FoldChange < -threshold,],
                    aes(y=-log10(padj), x=-threshold), shape = 2, colour="gray30") +
          xlim(-threshold,threshold) +
          scale_color_manual(values = c("gray30","#d83a00","#ff9b76","#ffd4c4")) +
          ylab(paste0("- log10(FDR)")) +
          xlab("Fold Change Nr6a1 vs Ctrl ATAC") + 
          ggtitle(paste0("Results Nr6a1 vs Ctrl")) +
          theme_bw() 



plot_nr6a1ATAC_volcano <- ggplot(results_sub_plot1, aes(x=log2FoldChange, y=-log10(padj), color=color_sig)) +
          rasterise(geom_point(size=1),dpi = 300) +
          geom_point(data=results_sub_plot1[results_sub_plot1$log2FoldChange > threshold,],
                   aes(y=-log10(padj), x=threshold), shape = 2, colour="gray30") +
          geom_point(data=results_sub_plot1[results_sub_plot1$log2FoldChange < -threshold,],
                    aes(y=-log10(padj), x=-threshold), shape = 2, colour="gray30") +
          xlim(-threshold,threshold) +
          scale_color_manual(values = c("gray30","#d83a00","#ff9b76","#ffd4c4")) +
          ylab(paste0("- log10(FDR)")) +
          xlab("Fold Change Nr6a1 vs Ctrl ATAC") + 
          ggtitle(paste0("Results Nr6a1 vs Ctrl")) +
          theme_bw() 


ggsave(paste0(workingdir,outdir,"Volcano_Nr6a1OE_ATAC.pdf"), plot=plot_nr6a1ATAC_volcano,
             width=4, height=3, units="in", useDingbats=FALSE)




```


Export table with the significant peaks and closest genes 

```{r}
## Annotation table
ann_table <- read.table(file=paste0(workingdir,"inputs_glialnr6a1_2/","consensus_peaks.mLb.clN.annotatePeaks.txt"), header=TRUE, stringsAsFactors = FALSE, sep = "\t")
colnames(ann_table)[1] <- "Peakid"

ann_table_clean <- ann_table %>% 
  select(c("Peakid","Chr","Start","End","Strand","Annotation","Distance.to.TSS","Nearest.PromoterID")) %>%
  separate(Annotation, into = "Annotation_brief", sep = " ", remove = FALSE)


write.table(results_sub_plot1 %>% rownames_to_column("Peakid") %>% left_join(ann_table_clean, by = "Peakid"),
  file = paste0(workingdir,outdir,"Results_annotated_Nr6a1OE-ATAC_differentconsensus.txt"),
  sep = "\t", quote = FALSE, row.names = FALSE)

```


Extract the bed files 

```{r}

# this is zeo
Nr6a1_up <- dds_results %>% 
  as.data.frame() %>%
  filter(padj < 0.001 & log2FoldChange > 0) %>%
  rownames_to_column("Peakid")

Nr6a1_down <- dds_results %>% 
  as.data.frame() %>%
  filter(padj < 0.001 & log2FoldChange < 0) %>%
  rownames_to_column("Peakid")


Nr6a1_down_bed <- ann_table_clean %>%
  select(Chr,Start,End, Strand,Peakid) %>%
  filter(Peakid %in% Nr6a1_down$Peakid)

# write.csv(Foxa2_up, file="1_outputs_DESeq2results/Foxa2_up_results.csv", row.names = F, quote = F)
# write.csv(Foxa2_down, file="1_outputs_DESeq2results/Foxa2_down_results.csv", row.names = F, quote = F)
# 
# 
write.table(Nr6a1_down_bed, file=paste0(workingdir,outdir,"Nr6a1_down.bed"),sep="\t", row.names = F, col.names = F, quote = F)

```



```{r}
sessionInfo()
```
